sphinx_gallery/scrapers.py

```diff
@@ -91,8 +102,37 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
         the images. This is often produced by :func:`figure_rst`.
     """
     matplotlib, plt = _import_matplotlib()
+    from matplotlib.animation import FuncAnimation, ImageMagickWriter
     image_path_iterator = block_vars['image_path_iterator']
     image_rsts = []
+    # Animations first
+    if gallery_conf.get('matplotlib_animations', False):
+        for ani in block_vars['example_globals'].values():
+            if isinstance(ani, FuncAnimation) and \
+                    not hasattr(ani, '_sg_scraped'):
+                ani._sg_scraped = True  # ugly monkeypatch, but should work
```

Comment:
Maybe this should be last instead of first?

Reply:
Maybe this means that animations are the thumbnail by default as they are the first 'image'?

Reply:
Good catch, this definitely did not preserve order (bad). Fixing in this commit